packages/eslint-plugin/docs/rules/no-unsafe-enum-comparison.md

```diff
@@ -0,0 +1,70 @@
+---
+description: 'Disallow comparing an enum value with a non-enum value.'
+---
+
+> 🛑 This file is source code, not the primary documentation location! 🛑
+>
+> See **https://typescript-eslint.io/rules/no-unsafe-enum-comparison** for documentation.
+
+The TypeScript compiler can be surprisingly lenient when working with enums.
+For example, it will allow you to compare enum values against numbers even though they might not have any type overlap:
+
+```ts
+enum Fruit {
+  Apple,
+  Banana,
+}
+
+let fruit = Fruit.Apple;
+fruit === 999; // No error
+```
+
+This rule flags when an enum typed value is compared to a non-enum `number`.
+
+<!--tabs-->
+
+### ❌ Incorrect
+
+```ts
+enum Fruit {
+  Apple,
+}
+
+let fruit = Fruit.Apple;
+fruit === 999;
+```
+
+```ts
+enum Vegetable {
+  Asparagus = 'asparagus',
+}
+
+let vegetable = Vegetable.Asparagus;
+vegetable === 'asparagus';
+```
+
+### ✅ Correct
+
+```ts
+enum Fruit {
+  Apple,
+}
+
+let fruit = Fruit.Apple;
+fruit === Fruit.Banana;
+```
+
+```ts
+enum Vegetable {
+  Asparagus = 'asparagus',
+}
+
+let vegetable = Vegetable.Asparagus;
+vegetable === Vegetable.Asparagus;
+```
```

Comment:
rule seem to not be working as described

```ts
enum Vegetable {
  Asparagus = 'asparagus',
  Onion = 'onion',
}

let vegetable: Vegetable = Vegetable.Asparagus

vegetable === Vegetable.Asparagus;
vegetable === Vegetable.Onion; // Errors
```

```ts
enum Vegetable {
  Asparagus,
  Onion,
}

let vegetable: Vegetable = Vegetable.Asparagus

vegetable === Vegetable.Asparagus;
vegetable === Vegetable.Onion; // Errors
```

------

we should either correct docs or change how rule works:

-------

```ts
enum Fruit {
  Apple,
}
let fruit = Fruit.Apple;
fruit === Fruit.Banana;
```

`Banana` is not defined

Reply:
Oh ha yeah the vegetable ones are from TypeScript immediately widening the variable from the initial value...